### PR TITLE
docs(robot-server): emulator and simulator notes.

### PR DESCRIPTION
**Pull request by amitlissack**
# Overview

blurb about running in dev modes

# Changelog

docs

# Review requests

# Risk assessment

None

### CODE REVIEW

File: robot-server/README.rst [hunk 1]
```diff
@@ -52,3 +52,27 @@ Tests should be organized similarly to the organization of the module itself.
...
 New code should have appropriate type annotations, and refactors of old code should try to add type annotations. We’re flexible about the refactor part, though - if adding type annotations greatly expands the scope of a PR, it’s OK to not add them as long as you explain this in the PR message.
+
+Developer Modes
+-----------------
+
+The robot server can be run on a PC in one of two development modes.
+
+These can be useful when an OT-2 and modules are not available.
+
+The **Opentrons** application will automatically discover a locally running robot server as **dev**.
```
Comment (SyntaxColoring) [inline robot-server/README.rst:63]: Does this not require you to add `localhost` to **More > Network & System > Manually Add Robot Network Addresses**, or have I been doing it wrong this whole time?

File: robot-server/README.rst [hunk 2]
```diff
@@ -52,3 +52,27 @@ Tests should be organized similarly to the organization of the module itself.
...
+
+***************
+Simulators
+***************
+`make dev` or will run the robot-server with simple software simulations of the Smoothie and magnetic, temperature, and thermocycler modules.
```
Comment (SyntaxColoring) [inline robot-server/README.rst]: Looks like an extra `or`?

File: robot-server/README.rst [hunk 3]
```diff
@@ -52,3 +52,27 @@ Tests should be organized similarly to the organization of the module itself.
...
+
+***************
+Emulators
+***************
+Using the emulation mode will have the robot server communicate to the emulation application. In this mode, the robot server is running as it would on the OT-2.
```
Comment (SyntaxColoring) [inline robot-server/README.rst]: It's not clear to me from these descriptions what the difference between emulation and simulation is.

I think it's:

* Simulation is higher-level.
* Emulation is the more thorough one. G-code commands and simulated responses get sent over a simulated serial connection.

Right?

Comment (amitlissack) [reply robot-server/README.rst]: That is correct.

**Review by celsasser — APPROVED** [review]
Very helpful!

**Review by SyntaxColoring — APPROVED** [review]
Changes clarified things for me. Thanks!

File: robot-server/README.rst (hunk 1 above)
Comment (amitlissack) [reply robot-server/README.rst:63]: That step is not required.

**Pull request merged**